### PR TITLE
Linekick with preparatory move

### PR DESCRIPTION
**Pull request by CollinAvidano**
Added a preparatory move subbehavior into line kick to get into position before it is enabled. Skill culling skills that did this manually

### CODE REVIEW

File: soccer/gameplay/plays/restarts/our_free_kick.py [hunk 1]
```diff
@@ -67,12 +66,11 @@ def score(cls):
             gs.is_ready_state() and gs.is_our_free_kick()) else float("inf")
 
     def receiver_near_pos(self):
-        return len(main.our_robots()) <= 4 or (self.subbehavior_with_name('receiver').robot is not None and \
-               (self.subbehavior_with_name('receiver').robot.pos - self.pos_up_field).mag() < 0.5)
+        return len(main.our_robots()) <= 4 or (self.has_subbehavior_with_name('receiver') and (self.subbehavior_with_name('receiver').robot is not None)) and \
```
Comment (tjones320) [inline soccer/gameplay/plays/restarts/our_free_kick.py]: We need to clean this up later

File: soccer/gameplay/plays/restarts/our_free_kick.py [hunk 2]
```diff
@@ -81,27 +79,15 @@ def on_enter_move(self):
...
-        self.add_subbehavior(skills.move.Move(self.pos_up_field), 'receiver', required=False, priority = 5)
+        if self.indirect and (self.receive_value == 0 or len(main.our_robots()) <= 5):
+            self.add_subbehavior(skills.move.Move(self.pos_up_field), 'receiver', required=False, priority = 5)
 
     def execute_move(self):
         self.move_pos = self.calc_move_pos()
 
-    def on_exit_move(self):
-        self.remove_all_subbehaviors()
+    # def on_exit_move(self):
```
Comment (tjones320) [inline soccer/gameplay/plays/restarts/our_free_kick.py]: We need to remove these comments

File: soccer/gameplay/skills/line_kick.py [hunk 3]
```diff
@@ -44,6 +45,13 @@ def __init__(self):
         self.max_speed = None
         self.max_accel = None
 
+    def on_enter_waiting(self):
+        super().recalculate_aim_target_point()
```
Comment (tjones320) [inline soccer/gameplay/skills/line_kick.py]: long term we need to use move_to if we are going to keep this

File: soccer/gameplay/skills/line_kick.py [hunk 4]
```diff
@@ -74,15 +82,16 @@ def execute_kick(self):
...
-        if main.ball().valid:
-            reqs.destination_shape = main.ball().pos
-        reqs.require_kicking = True
-        if self.use_chipper:
-            reqs.chipper_preference_weight = role_assignment.PreferChipper
-
-        if self.shell_id:
-            reqs.required_shell_id = self.shell_id
-        return reqs
+    # TODO Figure out how to renable role requirements for this however it needs to be done
```
Comment (tjones320) [inline soccer/gameplay/skills/line_kick.py]: if you use move_to you should be able to re-enable it

File: soccer/gameplay/tactics/one_touch_pass.py [hunk 5]
```diff
@@ -36,19 +36,19 @@ def __init__(self,
         self.force_reevauation = False
 
         rp = self.calc_receive_point()
-        self.add_subbehavior(skills.move.Move(main.ball().pos - (rp - main.ball().pos).normalized()*.4), 'passSetup', priority=5)
-        self.add_subbehavior(skills.move.Move(rp), 'receiveSetup', priority=5)
+        # self.add_subbehavior(skills.move.Move(main.ball().pos - (rp - main.ball().pos).normalized()*.4), 'passSetup', priority=5)
```
Comment (tjones320) [inline soccer/gameplay/tactics/one_touch_pass.py]: remove comments

File: soccer/gameplay/tactics/one_touch_pass.py [hunk 6]
```diff
@@ -36,19 +36,19 @@ def __init__(self,
...
 
         self.add_transition(behavior.Behavior.State.start,
-                            OneTouchPass.State.setup, lambda: True,
+                            OneTouchPass.State.passing, lambda: True,
                             'immediately')
 
-        self.add_transition(OneTouchPass.State.setup,
-                            OneTouchPass.State.passing, lambda: self.subbehavior_with_name('passSetup').is_done_running(),
-                            'movedKicker')
+        # self.add_transition(OneTouchPass.State.setup,
```
Comment (tjones320) [inline soccer/gameplay/tactics/one_touch_pass.py]: remove comments eventually

File: soccer/gameplay/tactics/one_touch_pass.py [hunk 7]
```diff
@@ -111,8 +111,8 @@ def reset_receive_point(self):
 
 
     def on_enter_passing(self):
-        self.remove_subbehavior('passSetup')
-        self.remove_subbehavior('receiveSetup')
+        # self.remove_subbehavior('passSetup')
```
Comment (tjones320) [inline soccer/gameplay/tactics/one_touch_pass.py]: remove comment eventually...

File: soccer/gameplay/plays/restarts/our_free_kick.py [hunk 8]
```diff
@@ -35,22 +35,22 @@ def __init__(self, indirect=None):
...
 
-        if (main.ball().pos.y > constants.Field.Length / 2):
-            self.indirect = gs.is_indirect()
-        else:
-            self.indirect = False
+        self.indirect = gs.is_indirect()
+        # if (main.ball().pos.y > constants.Field.Length / 2):
+        #     self.indirect = gs.is_indirect()
+        # else:
+        #     self.indirect = False
```
Comment (JNeiger) [inline soccer/gameplay/plays/restarts/our_free_kick.py]: Undo these

File: soccer/gameplay/skills/line_kick.py (hunk 3 above)
Comment (JNeiger) [reply soccer/gameplay/skills/line_kick.py]: We should just make one above that uses both the move and line kick subbehavior. You dont what this move in there when there are moving balls and that.

**Pull request merged**